### PR TITLE
ELE-2398 - move sort_alerts from API to data monitoring alerts flow

### PR DESCRIPTION
**Pull request by IDoneShaveIt**
The idea behind this change is that the alerts API should not be aware of the monitoring logic for filtering / skipping alerts.
This makes much more sense + it will help the SaaS alerts be more aligned with the OSS alerts code 🙂 

The main changes here is moving code from the alerts API to the DataMonitoringAlerts flow and updating tests.

### CODE REVIEW

File: elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py [hunk 1]
```diff
@@ -82,17 +80,88 @@ def _populate_data(
...
-            skip=filter_alerts(alerts=data.skip, alerts_filter=self.selector_filter),
+    def _filter_data(self, data: List[PendingAlertSchema]) -> List[PendingAlertSchema]:
+        return filter_alerts(data, alerts_filter=self.selector_filter)
+
+    def _fetch_last_sent_times(self, days_back: int) -> Dict[str, datetime]:
+        return self.alerts_api.get_alerts_last_sent_times(
+            days_back=days_back,
         )
 
+    def _sort_alerts(
```
Comment (IDoneShaveIt) [inline elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py:96]: Same code as it was on the alerts API

File: elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py [hunk 2]
```diff
@@ -82,17 +80,88 @@ def _populate_data(
...
+            if (
+                valid_alert.id in suppressed_alerts
+                or valid_alert.id not in latest_alert_ids
+            ):
+                alerts_to_skip.append(valid_alert)
+            else:
+                alerts_to_send.append(valid_alert)
+        return SortedAlertsSchema(send=alerts_to_send, skip=alerts_to_skip)
+
+    def _get_suppressed_alerts(
```
Comment (IDoneShaveIt) [inline elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py:116]: Same code as it was on the alerts API

File: elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py [hunk 3]
```diff
@@ -82,17 +80,88 @@ def _populate_data(
...
+                if last_sent_time
+                else False
+            )
+            if is_alert_in_suppression:
+                suppressed_alerts.append(alert.id)
+
+        return suppressed_alerts
+
+    @staticmethod
+    def _get_latest_alerts(
```
Comment (IDoneShaveIt) [inline elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py:142]: Same code as it was on the alerts API

**Review by NoyaArie — APPROVED** [review]
LGTM 🎉

**Pull request merged**